sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ModelTests.cs

```diff
@@ -173,18 +244,18 @@ public void ThrowsIfWireIsNotJson()
         {
             if (ModelInstance is IJsonModel<T> jsonModel && IsXmlWireFormat)
             {
-                Assert.Throws<InvalidOperationException>(() => jsonModel.Write(new Utf8JsonWriter(new MemoryStream()), _wireOptions));
+                Assert.Throws<FormatException>(() => jsonModel.Write(new Utf8JsonWriter(new MemoryStream()), _wireOptions));
```

Comment:
We always throw FormatException when the format is not expected (see the above test case)
This case could previously pass because we always has the flag `IsXmlWireFormat` false.